public/app/features/scenes/variables/sets/SceneVariableSet.ts

```diff
@@ -131,24 +130,54 @@ export class SceneVariableSet extends SceneObjectBase<SceneVariableSetState> imp
    * This will trigger an update of all variables that depend on it.
    * */
   private onVariableValueChanged = (event: SceneVariableValueChangedEvent) => {
-    const variable = event.payload;
+    const variableThatChanged = event.payload;
+
+    this.variablesThatHaveChanged.add(variableThatChanged);
 
     // Ignore this change if it is currently updating
-    if (this.updating.has(variable.state.name)) {
+    if (this.updating.has(variableThatChanged)) {
       return;
     }
 
-    for (const [name, deps] of this.dependencies) {
-      if (deps.includes(variable.state.name)) {
-        const otherVariable = this.getByName(name);
-        if (otherVariable) {
-          this.variablesToUpdate.set(name, otherVariable);
+    // Add variables that depend on the changed variable to the update queue
+    for (const otherVariable of this.state.variables) {
+      if (otherVariable.variableDependency) {
+        if (otherVariable.variableDependency.hasDependencyOn(variableThatChanged.state.name)) {
+          this.variablesToUpdate.add(otherVariable);
         }
       }
     }
 
     this.updateNextBatch();
   };
+
+  /**
+   * Walk scene object graph and update all objects that depend on variables that have changed
+   */
+  private notifyDependentSceneObjects() {
+    if (!this.parent) {
+      return;
+    }
+
+    this.traverseSceneAndNotify(this.parent);
```

Comment:
Just to make sure I understand this correctly - by calling this against the parent, we are making sure that children defined on the same level as variables will get notified too, not only objects below?

Reply:
yes exactly. since $variables is a branch in the scene graph, but really it's the parent of $variables that we need to traverse 

```ts
 const scene = new SceneFlexLayout({
          $variables: new SceneVariableSet({ variables: [B, A] }),
          children: [helloText, sceneObjectWithVariable],
        });
```

this = SceneVariableSet , and we want to notify the tree starting from it's parent (The flex layout in the above example)